### PR TITLE
Add prof command to add professors to contacts

### PR DESCRIPTION
**Pull request by ChrysanthemumT**
this change broke almost all the test, fixing it was a pain. :(
don't want to add more test at the moment, will rectify sometime in the future.
also adding profs is a little weird at the moment, couldn't get fuzzy finding with Levenshtein distance to work,
so followed similar logic to find to find profs in the data, which means that chinese names like "rui en" will add all profs with rui and en in their names. 
could maybe be improved in the future.
also adding images might be cool in the futre

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddProfCommand.java
Comment (cyqjoseph) [inline src/main/java/seedu/address/logic/commands/AddProfCommand.java:1]: Great add!

File: src/main/java/seedu/address/MainApp.java [hunk 1]
```diff
@@ -73,22 +77,29 @@ public void init() throws Exception {
...
         ReadOnlyAddressBook initialData;
+        ReadOnlyAddressBook profData;
         try {
             addressBookOptional = storage.readAddressBook();
+            profDataOptional = storage.readProfData();
             if (!addressBookOptional.isPresent()) {
                 logger.info("Creating a new data file " + storage.getAddressBookFilePath());
             }
             initialData = addressBookOptional.orElseGet(() -> new AddressBook());
+            profData = profDataOptional.orElseGet(() -> new ProfData());
```
Comment (jerremyng) [inline src/main/java/seedu/address/MainApp.java:94]: Good use of optionals!

File: src/main/java/seedu/address/logic/parser/AddProfCommandParser.java
Comment (cyqjoseph) [inline src/main/java/seedu/address/logic/parser/AddProfCommandParser.java:1]: Remember to add test coverage here!

File: src/test/java/seedu/address/logic/commands/AddAliasCommandTest.java
Comment (cyqjoseph) [inline src/test/java/seedu/address/logic/commands/AddAliasCommandTest.java:1]: Great work fixing up the test files!

File: src/main/java/seedu/address/model/person/Phone.java [hunk 2]
```diff
@@ -11,7 +11,7 @@ public class Phone {
 
     public static final String MESSAGE_CONSTRAINTS =
             "Phone numbers should only contain numbers, and it should be at least 3 digits long";
-    public static final String VALIDATION_REGEX = "\\d{3,}";
+    public static final String VALIDATION_REGEX = "^\\d*$";
```
Comment (jerremyng) [inline src/main/java/seedu/address/model/person/Phone.java:14]: We have to double check and make sure no bugs arise from regex change

File: src/main/java/seedu/address/storage/JsonProfDataStorage.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        Optional<JsonSerializableProfData> jsonProfData = JsonUtil.readJsonFile(
+                Paths.get(this.getClass().getResource(filePath.toString()).getPath()), JsonSerializableProfData.class);
+        if (!jsonProfData.isPresent()) {
+            return Optional.empty();
+        }
+        try {
+            return Optional.of(jsonProfData.get().toModelType());
+        } catch (IllegalValueException ive) {
+            logger.info("Illegal values found in " + filePath + ": " + ive.getMessage());
+            throw new DataLoadingException(ive);
```
Comment (jerremyng) [inline src/main/java/seedu/address/storage/JsonProfDataStorage.java:70]: Some test cases would be great!

**Review by cyqjoseph — APPROVED** [review]
Overall great addition! Users can differenciate between Professors now!

**Review by jerremyng — COMMENTED** [review]
Overall very good work done! Some test cases should make this a fantastic commit!

**Pull request merged**